### PR TITLE
Fix docs levels and broken links (without rename)

### PR DESCRIPTION
**Pull request by edenlightning**
Fix missing levels and links.

### CODE REVIEW

File: .github/BECOMING_A_CORE_CONTRIBUTOR.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ There's also [GitHub discussions](https://github.com/Lightning-AI/lightning/disc
 ## Applying
 
 There are no precise targets for becoming a core contributor. In the past, community members have become core after fitting the previous expectations consistently.
-We are on the lookout for new people to join, however, if you feel like you meet the expectations already and we haven't reached out to you yet, feel free to ping us privately on [![Discord](https://img.shields.io/discord/1077906959069626439?style=plastic)](https://discord.gg/VptPCZkGNa)!.
+We are on the lookout for new people to join, however, if you feel like you meet the expectations already and we haven't reached out to you yet, feel free to ping us privately on [Discord](https://discord.gg/VptPCZkGNa).
```
Comment (lantiga) [inline .github/BECOMING_A_CORE_CONTRIBUTOR.md:61]: Just to make sure: the Discord invite is permanent right?

**Review by lantiga — APPROVED** [review]
LGTM

File: .github/BECOMING_A_CORE_CONTRIBUTOR.md (hunk 1 above)
Comment (edenlightning) [reply .github/BECOMING_A_CORE_CONTRIBUTOR.md]: it is now :)

**Review by Borda — COMMENTED** [review]
can we do it without the level number in titles as these indes are very subjective and in the end inconsistent with the page names...

**Review by awaelchli — APPROVED** [review]

**Review by lantiga — APPROVED** [review]

File: docs/source-pytorch/expertise_levels.rst [hunk 2]
```diff
@@ -158,63 +158,63 @@ Configure all aspects of Lightning for advanced usecases.
...
+   :header: Level 18: Explore IPUs
    :description: Explore Intelligence Processing Unit (IPU) for model scaling.
    :col_css: col-md-6
    :button_link: levels/advanced_level_19.html
    :height: 150
    :tag: advanced
 
 .. displayitem::
-   :header: Level 20: Explore HPUs
+   :header: Level 19: Explore HPUs
```
Comment (Borda) [inline docs/source-pytorch/expertise_levels.rst:201]:
```suggestion
   :header: Explore HPUs
```

**Review by Borda — CHANGES_REQUESTED** [review]
@edenlightning could you pls remove level numbers from all titles where the level number does not have any value, for example, it is not clear why IPU shall have a lower index then TPU, etc.

**Comment by edenlightning** [comment]
> @edenlightning could you pls remove level numbers from all titles where the level number does not have any value, for example, it is not clear why IPU shall have a lower index then TPU, etc.

Will leave that up to @williamFalcon

**Pull request merged**